DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -115,268 +136,232 @@ class TTTrack {
-  theChi25Par = 0.0;
-  theStubPtConsistency4Par = 0.0;
-  theStubPtConsistency5Par = 0.0;
-  valid4ParFit = false;
-  valid5ParFit = false;
+  theMomentum = GlobalVector(0.0, 0.0, 0.0);
+  theRInv = 0.0;
+  thePOCA = GlobalPoint(0.0, 0.0, 0.0);
+  thePhiSector = 0;
+  theTrackSeed = 0;
+  theChi2 = 0.0;
+  theStubPtConsistency = 0.0;
+  NumFitPars = 0;
 }
 
 /// Another Constructor
 template <typename T>
 TTTrack<T>::TTTrack(std::vector<edm::Ref<edmNew::DetSetVector<TTStub<T> >, TTStub<T> > > aStubs) {
   theStubRefs = aStubs;
-  theMomentum4Par = GlobalVector(0.0, 0.0, 0.0);
-  theMomentum5Par = GlobalVector(0.0, 0.0, 0.0);
-  theRInv4Par = 0.0;
-  theRInv5Par = 0.0;
-  thePOCA4Par = GlobalPoint(0.0, 0.0, 0.0);
-  thePOCA5Par = GlobalPoint(0.0, 0.0, 0.0);
-  theSector = 0;
-  theWedge = 0;
-  theChi24Par = 0.0;
-  theChi25Par = 0.0;
-  theStubPtConsistency4Par = 0.0;
-  theStubPtConsistency5Par = 0.0;
-  valid4ParFit = false;
-  valid5ParFit = false;
+  theMomentum = GlobalVector(0.0, 0.0, 0.0);
+  theRInv = 0.0;
+  thePOCA = GlobalPoint(0.0, 0.0, 0.0);
+  thePhiSector = 0;
+  theTrackSeed = 0;
+  theChi2 = 0.0;
+  theStubPtConsistency = 0.0;
+  NumFitPars = 0;
+}
+
+/// Meant to be default constructor
+template <typename T>
+TTTrack<T>::TTTrack(double aRinv,
+                    double aphi,
+                    double aeta,
+                    double az0,
+                    double ad0,
+                    double aChi2,
+                    unsigned int aHitPattern,
+                    unsigned int nPar,
+                    double aBfield) {
+  theStubRefs.clear();
+  double thePT = 0.3 * aRinv * aBfield;
+  theMomentum = GlobalVector(GlobalVector::Cylindrical(thePT, aphi, thePT * sinh(aeta)));
+  theRInv = aRinv;
+  thePOCA = GlobalPoint(ad0 * cos(aphi), ad0 * sin(aphi), az0);
+  thePhiSector = 0;  // must be set externally
+  theTrackSeed = 0;  // must be set externally
+  theChi2 = aChi2;
+  theStubPtConsistency = 0.0;  // must be set externally
+  NumFitPars = nPar;
+  theHitPattern = aHitPattern;
+  theBField = aBfield;
+  // should probably fill the momentum vectur
 }
 
 /// Destructor
 template <typename T>
 TTTrack<T>::~TTTrack() {}
 
 template <typename T>
-void TTTrack<T>::setMomentum(GlobalVector aMomentum, unsigned int nPar) {
-  if (!checkValidArgsForSet(nPar))
-    return;
-
-  if (nPar == 4) {
-    valid4ParFit = true;
-    theMomentum4Par = aMomentum;
-  }
-
-  if (nPar == 5) {
-    valid5ParFit = true;
-    theMomentum5Par = aMomentum;
-  }
+void TTTrack<T>::setFitParNo(unsigned int nPar) {
+  NumFitPars = nPar;
 
   return;
 }
 
+// Note that these calls return the floating point values.  If a TTTrack is made with only ditized values,
+// the unpacked values must come from the TTTrack_Trackword member functions.
+
 template <typename T>
-GlobalVector TTTrack<T>::getMomentum(unsigned int nPar) const {
-  if (!checkValidArgs(nPar))
+GlobalVector TTTrack<T>::momentum() const {
+  if (NumFitPars == 5 || NumFitPars == 4) {
```

Comment:
4 and 5 should be named constants (as mentioned above)

Reply:
fixed.